doc/classes/vector.adoc

```diff
@@ -0,0 +1,264 @@
+= Vector API
+include::../../include/config.adoc[]
+:description: Documentation for the Vector class and API
+:keywords: mod, library, vector, x, y, z, pos, position
+
+TIP: The respective source code is located https://github.com/minetest/minetest/blob/master/builtin/common/vector.lua[here].
+
+== `vector` Namespace
+
+NOTE: The `vector` type used to be a simple `table` with `x`, `y`, and `z` values.
+However it has more recently been given metatable methods for convenience.
+Unless otherwise noted, the `vector` type always refers to the metatable-enhanced variety.
+Do note that functions here will accept an old-style (non-metatable) `vector`, but you cannot perform metatable operations with said `vector`.
+
+=== `vector.new(a, b, c) -> vector`
+* `a`: `number`, `vector`, or `nil`
+* `b`, `c`: `number` or `nil`
+
+If `a`, `b`, and `c` are `number`:: Returns a new `vector` where `{x = a, y = b, z = c}`
+
+NOTE: {deprecated}
```

Comment:
This doesn't clarify what's deprecated

Reply:
How does it not?
> The following call(s) are deprecated, please don't use them as they will be removed in a future version of the engine.

Reply:
Then the macro is inaccurate. Should say "deprecated-following".